docs/source/notes/designing_new_logic.rst

```diff
@@ -0,0 +1,351 @@
+
+            commands {
+                open/d :: "open {d}" :: "opening {d}";
+            }
+        }
+    }
+
+Similarly, to open a locked door with a push button, it is necessary to have the player at the same room as the push
+button is. Then the door is paired with the button, and the two rooms that this door connects to each other should be
+declared (note: the door and the button can be located in two different rooms, see the second set of rules below).
+Also the door is locked and the button is unpushed. From the `code` section, we realized that the defined command for
+this state transition is "push {b}". After this action, the door is unlocked and open and the button is changed to
+pushed. The rest of the conditions (predicates) are unchanged. This process is presented in following example for two
+scenarios: a) the button and the door ar at the same room, b) the push button is in separate room than the door.
+
+.. code-block:: bash
+
+    rules {
+        lock/close/db   :: $at(P, r) & $at(b, r) & $pair(b, d) & $link(r', d, r) & $link(r, d, r') & pushed(b) & open(d) & free(r, r') & free(r', r) -> unpushed(b) & locked(d);
+        unlock/open/db  :: $at(P, r) & $at(b, r) & $pair(b, d) & $link(r', d, r) & $link(r, d, r') & unpushed(b) & locked(d) -> pushed(b) & open(d) & free(r, r') & free(r', r);
+
+        lock/close/d/b  :: $at(P, r) & $at(b, r) & $pair(b, d) & $link(r', d, r'') & $link(r'', d, r') & pushed(b) & open(d) & free(r', r'') & free(r'', r') -> unpushed(b) & locked(d);
+        unlock/open/d/b :: $at(P, r) & $at(b, r) & $pair(b, d) & $link(r', d, r'') & $link(r'', d, r') & unpushed(b) & locked(d) -> pushed(b) & open(d) & free(r', r'') & free(r'', r');
+    }
+
+    reverse_rules {
+        lock/close/d/b :: unlock/open/d/b;
+        lock/close/db  :: unlock/open/db;
+    }
+
+    inform7 {
+        ...
+
+        commands {
+            lock/close/d/b  :: "push {b}" :: "_pushing the {b}";
+            unlock/open/d/b :: "push {b}" :: "_pushing the {b}";
+
+            lock/close/db   :: "push {b}" :: "_pushing the {b}";
+            unlock/open/db  :: "push {b}" :: "_pushing the {b}";
+        }
+    }
+
+As this example illustrates, since the same command is used for an action in two different situations, the rule names
+are different, although the command and the inform7 acknowledgement are all the same.
+
+Have you noticed the reverse_rule section? In this section, the reverse rules are connected to each other to inform the
+framework that after taking an action what would be the reverse action to get back to the current state. This provides
+the possibility of getting back to a state after moving from it, also provides back and forth exploration within the
+environment.
+
+The last section of a logic file is the `constraints` which defines the failure rules; i.e. describes that which
+predicates cannot occur simultaneously in a state. This section is only required to be defined in TWLL. Following is an
```

Comment:
Constraints are only used during quest generation and game creation via GameMaker. It ensures the initial state is valid.